shared/macros-ansible.jinja

```diff
@@ -346,3 +346,195 @@ The macro requires following parameters:
     create: yes
   when: find_existing_watch_audit_rules.matched is defined and find_existing_watch_audit_rules.matched == 0
 {{%- endmacro %}}
+
+{{#
+The following macro remediates Audit syscall rule in /etc/audit/rules.d directory.
+The macro requires following parameters:
+- arch: must be 32 or 64, this distinguishes the architecture (32 bit or 64 bit). Rules for appropriate architecture will be used.
+- syscalls: list of syscalls supplied as a list ["syscall1", "syscall2"] etc.
+- key: a key to use as rule identifier.
+Note that if there  already exists a rule wit the same key in the /etc/audit/rules.d directory, the rule will be placed in the same file.
+#}}
+
+{{% macro ansible_audit_augenrules_add_syscall_rule(arch="", syscalls=[], key="") -%}}
+#
+# What architecture are we on?
+#
+- name: Set architecture for audit modules tasks
+  set_fact:
+    audit_arch: "b{{ ansible_architecture | regex_replace('.*(\\d\\d$)','\\1') }}"
+
+- name: Declare list of syscals
+  set_fact:
+    syscalls: {{{ syscalls }}}
+
+- name: Declare number of syscalls
+  set_fact: audit_syscalls_number_of_syscalls="{{ syscalls|length|int }}"
+
+- name: Check existence of syscalls for 32 bit architecture in /etc/audit/rules.d/
```

Comment:
The tasks for 32 bits are being included twice. As they are not wrapped in an `arch` conditional, when the call for b64 happens they are added too.

Reply:
This is done for a reason. Here is an excerpt from man page for audit.rules:
---
When you specify a syscall name, auditctl will look up the name and get its syscall number. This leads to some problems on bi-arch machines. The 32 and 64 bit syscall numbers sometimes, but not always, line up. So, to solve this problem, you would generally need to break the rule into 2 with one specifying −F arch=b32 and the other specifying −F arch=b64. This needs to go in front of the −S option so that auditctl looks at the right lookup table when returning the number.
---
I agree that it is not clear, it deserves some explanation. Of course other suggestions how to solve this are welcomed.

Reply:
I understand that the remediation needs to add audit rules for both 32 and 64 bit syscalls.
But the original remediation didn't have task "Check existence of syscalls for 32 bit architecture in /etc/audit/rules.d/" twice, as the one generated now.
This is happening because the macro is called twice, but "Check existence of syscalls for 32 bit architecture in /etc/audit/rules.d/" is not wrapped in a conditional.

Reply:
Another way of handling this would be to have one macro that handles both 32 and 64 bits at once, basically taking the original remediation and parametrizing `syscalls` and `key`.

One benefit of this is that tasks like `Set architecture for audit modules tasks` and `Set architecture for audit modules tasks` won't be duplicated.